### PR TITLE
SP2-657 Add ability to make a Goal as Achieved

### PR DESCRIPTION
**Pull request by pipwilson**
This PR adds two new controllers and two new pages. 

1) confirm that goal has been achieved
2) display achieved goal

Although it allows a Goal to marked as Achieved it does not include the ability to make a note at the same time - this will need to be added in a subsequent PR once SP2-594 is complete.

### CODE REVIEW

File: server/@types/NewGoalType.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
+import { GoalStatus } from './GoalType'
+
 export type NewGoal = {
-  title: string
-  areaOfNeed: string
+  title?: string
```
Comment (Thomas-Geraghty) [inline server/@types/NewGoalType.ts]: Doesn't a NewGoal always need a title?

Comment (pipwilson) [reply server/@types/NewGoalType.ts]: the matching API change ensures that there's a submitted title when creating a new one, but basically I wanted to be able to submit a partial NewGoal in order to do updates against a known UUID without having to retrieve and submit all the data again.

Comment (Thomas-Geraghty) [reply server/@types/NewGoalType.ts]: It might instead be better to use the `Partial<Goal>` type then, or `Partial<NewGoal>`, whichever your fancy

Comment (pipwilson) [reply server/@types/NewGoalType.ts]: done in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/147/commits/f17ff510a0cf97caeaabf4526cb4a7f6eea6de47

**Review by Thomas-Geraghty — APPROVED** [review]

**Pull request merged**